comp/core/gui/guiimpl/systray/Sources/WiFiIPCServer.swift

```diff
@@ -0,0 +1,249 @@
+/// Request structure for IPC commands
+struct WiFiIPCRequest: Codable {
+    let command: String
+}
+
+/// Response structure for IPC
+struct WiFiIPCResponse: Codable {
+    let success: Bool
+    let data: WiFiData?
+    let error: String?
+}
+
+/// WiFiIPCServer handles Unix socket communication for WiFi data
+class WiFiIPCServer {
+    private let wifiDataProvider: WiFiDataProvider
+    private var socketFileDescriptor: Int32 = -1
+    private var isRunning = false
+    private var acceptQueue: DispatchQueue
+    private let socketPath: String
+
+    init(wifiDataProvider: WiFiDataProvider) {
+        self.wifiDataProvider = wifiDataProvider
+
+        // Get installation path from environment (set by LaunchAgent) or use default
+        let installPath = ProcessInfo.processInfo.environment["DD_INSTALL_PATH"] ?? "/opt/datadog-agent"
+
+        // Create socket path based on current user's UID
+        let uid = getuid()
+        self.socketPath = "\(installPath)/run/gui-\(uid).sock"
+        // Use .background QoS for agent telemetry collection (not user-facing work)
+        self.acceptQueue = DispatchQueue(label: "com.datadoghq.wifi.ipc", qos: .background)
+
+        Logger.info("Initialized with socket path: \(socketPath)", context: "WiFiIPCServer")
+    }
+
+    /// Start the IPC server
+    func start() throws {
+        guard !isRunning else {
+            Logger.info("Server already running", context: "WiFiIPCServer")
+            return
+        }
+
+        // Create socket directory if it doesn't exist
+        try createSocketDirectory()
+
+        // Remove existing socket file if present
+        try? FileManager.default.removeItem(atPath: socketPath)
+
+        // Create Unix domain socket
+        socketFileDescriptor = socket(AF_UNIX, SOCK_STREAM, 0)
+        guard socketFileDescriptor >= 0 else {
+            throw NSError(domain: "WiFiIPCServer", code: 1, 
+                         userInfo: [NSLocalizedDescriptionKey: "Failed to create socket: \(String(cString: strerror(errno)))"])
+        }
+
+        // Set socket to non-blocking and reusable
+        var value: Int32 = 1
+        setsockopt(socketFileDescriptor, SOL_SOCKET, SO_REUSEADDR, &value, socklen_t(MemoryLayout<Int32>.size))
+
+        // Bind socket to path
+        var addr = sockaddr_un()
+        addr.sun_family = sa_family_t(AF_UNIX)
+
+        let pathBytes = socketPath.utf8CString
+        guard pathBytes.count <= MemoryLayout.size(ofValue: addr.sun_path) else {
+            close(socketFileDescriptor)
+            throw NSError(domain: "WiFiIPCServer", code: 2,
+                         userInfo: [NSLocalizedDescriptionKey: "Socket path too long"])
+        }
+
+        withUnsafeMutableBytes(of: &addr.sun_path) { ptr in
+            pathBytes.withUnsafeBytes { pathPtr in
+                ptr.copyMemory(from: pathPtr)
+            }
+        }
+
+        // Set umask to create socket with 0o660 permissions atomically
+        let oldUmask = umask(0o117)
```

Comment:
Comment says 0o660 permission but code 0o117. Which permissions should be used? Also what are they in human language and more importantly why?